src/cpp/src/sampler.cpp

```diff
@@ -755,8 +755,8 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
             continue;
 
         size_t num_running_sequences = sequence_group->num_running_seqs();
-        size_t actual_seq_len = sequence_group->get_num_scheduled_tokens(); // points to a token which needs to be sampled
-        size_t padded_amount_of_processed_tokens = std::max(actual_seq_len, batch_seq_len);
+        size_t actual_seq_len = sequence_group->get_seq_len_to_sample(); // points to a token which needs to be sampled
+        size_t padded_amount_of_processed_tokens = std::max(sequence_group->get_num_scheduled_tokens(), batch_seq_len);
```

Comment:
looks like currently, `batch_seq_len ` is always 1 (because we merge all sequences to a single input of [total amount of tokens, 1]) and we can drop `batch_seq_len` at all and always shift on `actual_seq_len`